src/ui/common/src/components/pages/data/index.tsx

```diff
@@ -20,13 +20,31 @@ type Props = {
   user: UserProfile;
 };
 
-const SearchBar = (options: DataPreviewInfo[], onChangeFn: (any) => void) => {
+const SearchBar = (
+  options: DataPreviewInfo[],
+  onChangeFn: (v: string) => void,
+) => {
   return (
     <Autocomplete
       sx={{ width: 300 }}
       options={options}
+      onInputChange={(_, val, reason) => {
+        if (reason === "clear") {
+          onChangeFn("")
+        }
+
+        onChangeFn(val)
+      }}
       freeSolo
-      getOptionLabel={dataCardName}
+      getOptionLabel={(option) => {
+        // When option string is invalid, non of 'options' will be selected
+        // and the component will try to directly render the input string.
+        // This check prevents applying `dataCardName` to the string.
+        if(typeof option === 'string') {
```

Comment:
Seems like we have a couple formatting issues here. Can we make sure to run the linter here please?